teuthology/openstack.py

```diff
@@ -0,0 +1,375 @@
+                            str(providers))
+
+
+    def image(self):
+        if self.provider == 'ovh':
+            return 'Ubuntu 14.04'
+        elif self.provider == 'redhat':
+            return 'ubuntu-server-14.04-x86_64'
+        elif self.provider == 'entercloudsuite':
+            return 'GNU/Linux Ubuntu Server 14.04 Trusty Tahr x64'
+
+    def flavor(self):
+        hint = {
+            'disk': 10, # GB
+            'ram': 1024, # MB
+            'cpus': 1,
+        }
+        if self.args.simultaneous_jobs > 25:
+            hint['ram'] = 30000 # MB
+        elif self.args.simultaneous_jobs > 10:
+            hint['ram'] = 7000 # MB
+        elif self.args.simultaneous_jobs > 3:
+            hint['ram'] = 4000 # MB
+
+        select = None
+        if self.provider == 'ovh':
+            select = '^(vps|eg)-'
+        return super(TeuthologyOpenStack, self).flavor(hint, select)
+
+    def net(self):
+        if self.provider == 'entercloudsuite':
+            return "--nic net-id=default"
+        else:
+            return ""
+
+    def get_user_data(self):
+        path = tempfile.mktemp()
+        template = open(self.user_data).read()
+        openrc = ''
+        for (var, value) in os.environ.iteritems():
+            if var.startswith('OS_'):
+                openrc += ' ' + var + '=' + value
+        if self.args.integration_tests:
+            run_tests = 'yes'
+        else:
+            run_tests = 'no'
+        log.debug("OPENRC = " + openrc +
+                  "RUN_TESTS = " + run_tests +
+                  "NWORKERS = " + str(self.args.simultaneous_jobs) +
+                  "PROVIDER = " + self.provider)
+        content = (template.
+                   replace('OPENRC', openrc).
+                   replace('RUN_TESTS', run_tests).
+                   replace('NWORKERS', str(self.args.simultaneous_jobs)).
+                   replace('PROVIDER', self.provider))
+        open(path, 'w').write(content)
+        log.debug("get_user_data: " + content + " written to " + path)
+        return path
+
+    def create_security_group(self):
+        try:
+            misc.sh("openstack security group show teuthology")
+            return
+        except subprocess.CalledProcessError:
+            pass
+        # TODO(loic): this leaves the teuthology vm very exposed
+        # it would be better to be very liberal for 192.168.0.0/16
+        # and 172.16.0.0/12 and 10.0.0.0/8 and only allow 80/8081/22
+        # for the rest.
+        misc.sh("""
+openstack security group create teuthology
+openstack security group rule create --dst-port 1:10000 teuthology
+#openstack security group rule create --dst-port 8081 teuthology # pulpito
+#openstack security group rule create --dst-port 443 teuthology # https
+#openstack security group rule create --dst-port 115 teuthology # sftp
+#openstack security group rule create --dst-port 80 teuthology # http
+#openstack security group rule create --dst-port 22 teuthology # ssh
```

Comment:
Do you intend to uncomment these, or remove them - or, if you want to leave them present but commented, why?